src/charm.py

```diff
@@ -51,10 +52,13 @@ class MicroCephCharm(sunbeam_charm.OSBaseOperatorCharm):
 
     _state = ops.framework.StoredState()
     service_name = "microceph"
+    storage = None  # StorageHandler
```

Comment:
Storagehandler implements ops.framework.Object, this gives it the ability to define it's own 'on' object (to define custom events for future extend-ability), it can subscribe to the charm events (storage events in our case) which helps keep the handling neat and tidy as well.